### PR TITLE
Issue 1396 bad png path

### PR DESCRIPTION
**Pull request by doutriaux1**

### CODE REVIEW

File: Packages/vcs/Lib/Canvas.py [hunk 1]
```diff
@@ -4812,6 +4812,9 @@ def png(self, file, width=None, height=None,
     a.plot(array)
     a.png('example')       # Overwrite a png file
 """
+        base = os.path.dirname(file)
+        if base!="" and not os.path.exists(base):
+          raise vcsError("Output path: %s does not exists" % base)
```
Comment (durack1) [inline Packages/vcs/Lib/Canvas.py]: @doutriaux1 "does not exists" -> "does not exist"

File: Packages/vcs/Lib/Canvas.py [hunk 2]
```diff
@@ -4812,6 +4812,9 @@ def png(self, file, width=None, height=None,
     a.plot(array)
     a.png('example')       # Overwrite a png file
 """
+        base = os.path.dirname(file)
+        if base != "" and not os.path.exists(base):
+            raise vcsError("Output path: %s does not exists" % base)
```
Comment (durack1) [inline Packages/vcs/Lib/Canvas.py]: @doutriaux1 "does not exists" -> "does not exist"

File: Packages/vcs/Lib/Canvas.py [hunk 3]
```diff
@@ -4812,6 +4812,9 @@ def png(self, file, width=None, height=None,
     a.plot(array)
     a.png('example')       # Overwrite a png file
 """
+        base = os.path.dirname(file)
```
Comment (aashish24) [inline Packages/vcs/Lib/Canvas.py]: Why not put this block of code in try catch instead of checking of the file exists? In general thats a better practice

Comment (doutriaux1) [inline Packages/vcs/Lib/Canvas.py]: Because it does not fail, that is actually the issue.

Comment (doutriaux1) [inline Packages/vcs/Lib/Canvas.py]: @aashish24 are you suggesting we create the subdirectory?

File: Packages/vcs/Lib/Canvas.py [hunk 4]
```diff
@@ -4812,6 +4812,9 @@ def png(self, file, width=None, height=None,
...
+        if base != "" and not os.path.exists(base):
+            raise vcsError("Output path: %s does not exist" % base)
         return self.backend.png(
```
Comment (aashish24) [inline Packages/vcs/Lib/Canvas.py]: Do you mean that backend.png does not throw an exception or any other error/

File: Packages/vcs/Lib/Canvas.py (hunk 3 above)
Comment (aashish24) [inline Packages/vcs/Lib/Canvas.py]: No, I was not.

**Comment by aashish24** [comment]
LGTM :+1:

**Pull request merged**